### PR TITLE
[node] add util.types.isBigInt64Array, isBigUint64Array …

### LINKED ISSUE #30471: BigInt nodejs typings support
Typescript 3.2 will contain support for BigInt and will be released this month (I assume probably during the next days?). BigInt was also enabled by default in Node v10.4. Are there any efforts or plans made to implement the BigInt type definitions for node 10 yet?

@SimonSchick @Flarna @rbuckton

### PR DESCRIPTION
**Pull request by Flarna**
...and isModuleNamespaceObject

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://nodejs.org/dist/latest-v11.x/docs/api/util.html#util_util_types_isbigint64array_value
https://nodejs.org/dist/latest-v11.x/docs/api/util.html#util_util_types_isbiguint64array_value
https://nodejs.org/dist/latest-v11.x/docs/api/util.html#util_util_types_ismodulenamespaceobject_value
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

fixes #30471

### CODE REVIEW

File: types/node/v10/node-tests.ts [hunk 1]
```diff
@@ -952,6 +952,10 @@ function bufferTests() {
         const teEncodeRes: Uint8Array = te.encode("TextEncoder");
 
         // util.types
+        let b: Boolean;
+        b = util.types.isBigInt64Array(15);
+        b = util.types.isBigUint64Array(15);
```
Comment (SimonSchick) [inline types/node/v10/node-tests.ts]: Can you add a type guard test here, I'm worried TS won't do the type assertion correctly here.

File: types/node/v10/node-tests.ts [hunk 2]
```diff
@@ -952,6 +952,10 @@ function bufferTests() {
         const teEncodeRes: Uint8Array = te.encode("TextEncoder");
 
         // util.types
+        let b: Boolean;
```
Comment (SimonSchick) [inline types/node/v10/node-tests.ts]: `boolean`.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/node/v10/node-tests.ts (hunk 2 above)
Comment (Flarna) [reply types/node/v10/node-tests.ts]: done

File: types/node/v10/node-tests.ts (hunk 1 above)
Comment (Flarna) [reply types/node/v10/node-tests.ts]: done in ts 3.2. specific test as the generic one doesn't know about `BigInt64Array` and therefore can't do a type guard.
During playing around I found that the type guard for `isBoxedPrimitive()` is actually not really asserting anything useful but I failed to repair it. Seems typescript can't type guard correct with unions of boxed types.

**Comment by Flarna** [comment]
@SimonSchick Anything left to do here from my side?

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**